locale/en/docs/guides/backpressuring-in-streams.md

```diff
@@ -0,0 +1,480 @@
+buffer during data transfer. When the recieving end of the transfer has complex
+operations, or is slower for whatever reason, there is a tendency for data from 
+the incoming source to begin to accumulate.
+
+To solve this problem, there must be a delegation system in place to ensure a
+smooth flow of data from one source to another. Different communities have
+resolved this issue uniquely to their programs, Unix pipes and TCP sockets are
+good examples of this, and often times is referred to as _flow control_. In
+Node.js, streams have been the adopted solution.
+
+The purpose of this guide is to further detail what backpressure is, and how
+exactly streams address this in Node.js' source code. The second part of
+the guide will introduce suggested best practices to ensure your application's
+code is safe and optimized when implementing streams.
+
+We assume a little familiarity with the general definition of
+[`backpressure`][], [`Buffer`][], and [`EventEmitters`][] in Node.js, as well as
+some experience with [`Streams`][]. If you haven't read through those docs,
+it's not a bad idea to take a look at the API documentation first, as it will
+help expand your understanding while reading this guide and for following along
+with examples.
+
+## The Problem With Data Handling
+
+In a computer system, data is transferred from one process to another through
+pipes, sockets, and signals. In Node.js, we find a similar mechanism called
+[`Streams`][]. Streams are great! They do so much for Node.js and almost every
+part of the internal codebase utilizes that module. As a developer, you
+are more than encouraged to use them too!
+
+```javascript
+const readline = require('readline');
+
+// process.stdin and process.stdout are both instances of Streams
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+rl.question('Why should you use streams? ', (answer) => {
+
+  console.log(`Maybe it's ${answer}, maybe it's because they are awesome! :)`);
+
+  rl.close();
+});
+```
+
+A good example of why the backpressure mechanism implemented through streams are
+a great optimization can be demonstrated by comparing the internal system tools
+from Node.js' [`Streams`][] implementation.
+
+In one scenario, we take a large file (approximately ~9gb) and compress it up
+using the familiar [`zip(1)`][] tool.
+
+```
+$ zip The.Matrix.1080p.mkv
+```
+
+While that will take a few minutes to compress, in another shell we may run
+a script that takes Node.js' module [`Zlib`][], that wraps around another compression tool, [`gzip(1)`][].
+
+```javascript
+const gzip = require('zlib').createGzip();
+const fs = require('fs');
+
+const inp = fs.createReadStream('The.Matrix.1080p.mkv');
+const out = fs.createWriteStream('The.Matrix.1080p.mkv.gz');
+
+inp.pipe(gzip).pipe(out);
+```
+
+While the first `zip` function will ultimately fail, Node will be able to handle 
```

Comment:
The reason why I asked is that I'm fairly certain `zip` will not encounter this problem. Unlike Node.js, `zip` and other UNIX utilities operate synchronously. What that means is that they only `read()` as much as they need at one time, and immediately `write()` it out afterwards. Since these two system calls are synchronous, there would not be any backpressure problems since by definition they only appear in an asynchronous system.